### ISSUE TITLE
better Idea for "getExistingUser" when using oAuth

### ISSUE THREAD

**Issue by Jumaron** [JitPackJoyride/lucia-adapter-edgedb#11]
Currently when using oAuth, following the Guide on Lucia Auths official website, there is one thing that can be quickly overlooked when working with this Adapter.
The getExistingUser is not working due to its nature which is non functional.
The function itself does:
-Use an Auth Key to authorize
-Get the user data

```
Currently you could use something like this: 

async function useUsersKey(googleId: string) {
	const query = e.select(e.UserKey, (userKey) => ({
		...e.UserKey['*'],
		user: { ...e.User['*'] }, // Fetch all properties of the related User
		filter: e.op(userKey.key_id, '=', 'google:' + googleId),
		limit: 1
	}));

	return query.run(db);
}
```

Its working but, we should either make something properly inbuild or add this problem to the docs, so people are prepared.

Still great work for the adapter 👌

**Comment by JitPackJoyride** [comment]
Hi! Thanks for this issue! I genuinely was not expecting much usage so I wasn't checking too much here, sorry for the late response!

Also, I was digging through the source code to see what the issue was, and I'm pretty sure I've debugged it. I'm not properly returning the `user_id` in my `getKey` implementation. It's required for `getExistingUser` as stated in the line linked below. Great catch, and glad you have a workaround for the time being. I'll try to ship this fix and your PR changes later today.

https://github.com/lucia-auth/lucia/blob/54f62cbb036eba44f8870daa16f9b7c64a062af1/packages/oauth/src/core/provider.ts#L22

**Comment by Jumaron** [comment]
Thanks for the response!

Sounds about right, when you pushed the PR, I would be more then happy to test it out, to give you confirmation if everything works.

**Comment by JitPackJoyride** [comment]
Hi, I'm releasing it as a minor update, let me know if it works.

If it doesn't, I'll release some patch fixes if needed.

**Comment by JitPackJoyride** [comment]
Closing because the fix was released in #14 with the 0.2.0 version of the library

Specifically the fix was: 03bf32ae93e2e4ce3e920c33482b5ee5eac2f11c
